### PR TITLE
Fix geometry and bounding box

### LINKED ISSUE #89: Polar assets have bad geometries and bounding boxes
As originally reported https://github.com/microsoft/PlanetaryComputer/discussions/121. [@Pete Gadomski](https://dev.azure.com/planetary-computer/Planetary%20Computer%20Components/_backlogs/backlog/Team%20Element%2084/Stories/?showParents=true&workitem=3603#) did some digging to confirm the issue:

Image

### PR DESCRIPTION
**Pull request by pjhartzell**
**Related Issue(s):** 
- Closes #89 

**Description:**
Builds item geometry from the MODIS tile grid location (horizontal and vertical tile numbers) and sinusoidal projection parameters. This replaces the XML boundary points as the geometry source, which was problematic in areas of high projection distortion, and provides tight geometry polygons that curve with projection distortions.

Examples. Red is existing geometry. Green is new geometry.
image

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [x] Example STAC Catalog has been updated to reflect changes, if applicable.
- [x] Changes are added to the [CHANGELOG](../CHANGELOG.md).

### CODE REVIEW

File: src/stactools/modis/metadata.py [hunk 1]
```diff
@@ -13,6 +14,31 @@
 from stactools.modis import utils
 from stactools.modis.constants import TEMPORALLY_WEIGHTED_PRODUCTS
 
+SIN_SPHERE_RADIUS = 6371007.181
+SIN_TILE_METERS = 1111950
+SIN_X_MIN = -20015109
+SIN_Y_MAX = 10007555
```
Comment (gadomski) [inline src/stactools/modis/metadata.py:20]: Is there a document that we can link to, in a comment, pointing to where these values came from?

File: src/stactools/modis/metadata.py [hunk 2]
```diff
@@ -260,6 +280,44 @@ def datetime(self) -> Optional[datetime.datetime]:
...
+                y = SIN_Y_MAX - (row + 0.5) * pixel_width - vtile * SIN_TILE_METERS
+                lat = math.degrees(y / SIN_SPHERE_RADIUS)
+                lon = math.degrees(
+                    x / (SIN_SPHERE_RADIUS * math.cos(math.radians(lat)))
+                )
+                if lat >= -90 and lat <= 90 and lon >= -180 and lon <= 180:
+                    lon_lat.append([lon, lat])
+            return lon_lat
+
+        tile_pixel_size = [k for k, v in SIN_TILE_PIXELS.items() if collection in v][0]
```
Comment (gadomski) [inline src/stactools/modis/metadata.py:317]:
```suggestion
        tile_pixel_size = next((k for k, v in SIN_TILE_PIXELS.items() if collection in v), None)
        if tile_pixel_size is None:
            raise ValueError(f"unsupported MODIS collection: {collection}")
```

**Review by gadomski — CHANGES_REQUESTED** [review]

File: src/stactools/modis/metadata.py (hunk 1 above)
Comment (pjhartzell) [reply src/stactools/modis/metadata.py:20]: Yep. Done.

File: src/stactools/modis/metadata.py (hunk 2 above)
Comment (pjhartzell) [reply src/stactools/modis/metadata.py:317]: Done.

**Review by gadomski — APPROVED** [review]

**Pull request merged**